docs/index.md

```diff
@@ -0,0 +1,106 @@
+# Cloud provider for Azure
```

Comment:
what's the difference between `README.md` and `index.md`?

Reply:
index.md is for `github pages` to generate a doc website

Reply:
thanks, I think we may also need such index page for csi drivers.